.ci_support/build_all.py

```diff
@@ -46,39 +51,49 @@ def build_all(recipes_dir, arch):
-                if "channel_sources" in specific_config:
-                    for row in specific_config["channel_sources"]:
-                        channels = [c.strip() for c in row.split(",")]
-                        if channels != ['conda-forge', 'defaults'] and \
-                                channels != ['conda-forge/label/cf201901', 'defaults']:
-                            print("Not a standard configuration of channel_sources. Building {} individually.".format(folder))
-                            conda_build.api.build([os.path.join(recipes_dir, folder)], config=get_config(arch, channels))
-                            built = True
-                            break
-                if not built:
-                    old_comp_folders.append(folder)
-                    continue
-        if not built:
-            new_comp_folders.append(folder)
-
-    if old_comp_folders:
-        print("Building {} with conda-forge/label/cf201901".format(','.join(old_comp_folders)))
-        channel_urls = ['local', 'conda-forge/label/cf201901', 'defaults']
-        build_folders(recipes_dir, old_comp_folders, arch, channel_urls)
-    if new_comp_folders:
-        print("Building {} with conda-forge/label/main".format(','.join(new_comp_folders)))
-        channel_urls = ['local', 'conda-forge', 'defaults']
-        build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
+            if platform == 'osx' and (
+                    'MACOSX_DEPLOYMENT_TARGET' in text or
+                    'MACOSX_SDK_VERSION' in text):
+                config = safe_load(text)
+
+                if 'MACOSX_DEPLOYMENT_TARGET' in config:
+                    for version in config['MACOSX_DEPLOYMENT_TARGET']:
+                        version = tuple([int(x) for x in version.split('.')])
+                        deployment_version = max(deployment_version, version)
+                if 'MACOSX_SDK_VERSION' in config:
+                    for version in config['MACOSX_SDK_VERSION']:
+                        version = tuple([int(x) for x in version.split('.')])
+                        sdk_version = max(deployment_version, version)
+
+    with open(variant_config_file, 'r') as f:
+        variant_text = ''.join(f.readlines())
+        variant_config = safe_load(variant_text)
+
+    if deployment_version != (0, 0):
+        deployment_version = '.'.join([str(x) for x in deployment_version])
+        print("Overriding MACOSX_DEPLOYMENT_TARGET to be ", deployment_version)
+        variant_config['MACOSX_DEPLOYMENT_TARGET'] = [deployment_version]
+
+    if sdk_version != (0, 0):
+        sdk_version = '.'.join([str(x) for x in sdk_version])
+        print("Overriding MACOSX_SDK_VERSION to be ", sdk_version)
+        variant_config['MACOSX_SDK_VERSION'] = [sdk_version]
+
+    with open(variant_config_file, 'w') as f:
+        new_file = safe_dump(variant_config)
+        f.write(new_file)
```

Comment:
This would remove preprocessing selectors like https://github.com/conda-forge/staged-recipes/blob/master/.ci_support/linux64.yaml#L11-L16